go/store/types/uuid.go

```diff
@@ -26,6 +27,21 @@ const (
 	uuidNumBytes = 16
 )
 
+// UUIDHashedFromValues generates a UUID from the first 16 byes of the hash.Hash
+// generated from serialized |vals|.
+func UUIDHashedFromValues(nbf *NomsBinFormat, vals ...Value) (UUID, error) {
```

Comment:
Crazy that something called "universally unique" is still 16x smaller than the hash keyspace